client/pom.xml

```diff
@@ -98,6 +98,13 @@
           <encoding>UTF-8</encoding>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-javadoc-plugin</artifactId>
+        <configuration>
+          <source>${java.level}</source>
+        </configuration>
+      </plugin>
```

Comment:
adding the plugin here implies that the Javadoc is *always* built, not only when the `javadoc` profile is enabled ...

Reply:
Not for me:

~~~
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 1 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 1 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO] 
[INFO] ---------------------< org.eclipse.hono:hono-core >---------------------
[INFO] Building Hono Core 1.0-M2-SNAPSHOT
[INFO] -------------------------------[ bundle ]-------------------------------
[INFO] 
[INFO] --- buildnumber-maven-plugin:1.4:create-timestamp (default) @ hono-core ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-no-snapshots) @ hono-core ---
[INFO] Skipping Rule Enforcement.
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-java-11) @ hono-core ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.2:prepare-agent (default-prepare-agent) @ hono-core ---
[INFO] argLine set to -javaagent:/home/jreimann/.m2/repository/org/jacoco/org.jacoco.agent/0.8.2/org.jacoco.agent-0.8.2-runtime.jar=destfile=/home/jreimann/Development/workspace-hono/hono/core/target/jacoco.exec
[INFO] 
[INFO] --- maven-dependency-plugin:3.0.2:unpack-dependencies (copy_demo_certs) @ hono-core ---
[INFO] org.eclipse.hono:hono-demo-certs:jar:1.0-M2-SNAPSHOT already exists in destination.
[INFO] 
[INFO] --- maven-resources-plugin:3.0.2:resources (default-resources) @ hono-core ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/jreimann/Development/workspace-hono/hono/core/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.0:compile (default-compile) @ hono-core ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-resources-plugin:3.0.2:testResources (default-testResources) @ hono-core ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 11 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.0:testCompile (default-testCompile) @ hono-core ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.0:test (default-test) @ hono-core ---
[INFO] Tests are skipped.
[INFO] 
[INFO] --- animal-sniffer-maven-plugin:1.17:check (verify_java8_compatibility) @ hono-core ---
[INFO] Checking unresolved references to org.codehaus.mojo.signature:java18:1.0
[INFO] 
[INFO] --- maven-dependency-plugin:3.0.2:unpack-dependencies (copy_legal_docs) @ hono-core ---
[INFO] Unpacking /home/jreimann/.m2/repository/org/eclipse/hono/hono-legal/1.0-M2-SNAPSHOT/hono-legal-1.0-M2-SNAPSHOT.jar to /home/jreimann/Development/workspace-hono/hono/core/target/classes/META-INF with includes "" and excludes "META-INF/**,checkstyle/**"
[INFO] 
[INFO] --- maven-bundle-plugin:3.5.0:bundle (default-bundle) @ hono-core ---
[INFO] 
[INFO] --- maven-source-plugin:3.0.1:jar-no-fork (attach-sources) @ hono-core ---
[INFO] Building jar: /home/jreimann/Development/workspace-hono/hono/core/target/hono-core-1.0-M2-SNAPSHOT-sources.jar
[INFO] 
[INFO] --- jacoco-extras:0.1.4:xml (build-enhanced-xml-report) @ hono-core ---
[INFO] Loading execution data file /home/jreimann/Development/workspace-hono/hono/core/target/jacoco.exec
[INFO] Analyzed bundle 'hono-core' with 69 classes
[WARNING] Classes in bundle 'hono-core' do no match with execution data. For report generation the same class files must be used as at runtime.
[WARNING] Execution data for class org/eclipse/hono/tracing/MultiMapInjectAdapter does not match.
[WARNING] Execution data for class org/eclipse/hono/util/ResourceIdentifier does not match.
[WARNING] Execution data for class org/eclipse/hono/util/Strings does not match.
[WARNING] Execution data for class org/eclipse/hono/config/KeyLoader does not match.
[WARNING] Execution data for class org/eclipse/hono/config/PemReader does not match.
[WARNING] Execution data for class org/eclipse/hono/util/MessageHelper does not match.
[WARNING] Execution data for class org/eclipse/hono/tracing/JsonObjectExtractAdapter$1 does not match.
[WARNING] Execution data for class org/eclipse/hono/auth/MessageDigestPasswordEncoder does not match.
[WARNING] Execution data for class org/eclipse/hono/util/JsonBackedValueObject does not match.
[WARNING] Execution data for class org/eclipse/hono/connection/impl/ConnectionFactoryImpl does not match.
[WARNING] Execution data for class org/eclipse/hono/auth/SpringBasedHonoPasswordEncoder does not match.
[WARNING] Execution data for class org/eclipse/hono/config/AbstractConfig does not match.
[WARNING] Execution data for class org/eclipse/hono/connection/ConnectTimeoutException does not match.
[WARNING] Execution data for class org/eclipse/hono/util/TimeUntilDisconnectNotification does not match.
[WARNING] Execution data for class org/eclipse/hono/util/CacheDirective does not match.
[WARNING] Execution data for class org/eclipse/hono/tracing/MessageAnnotationsInjectAdapter does not match.
[WARNING] Execution data for class org/eclipse/hono/tracing/MessageAnnotationsExtractAdapter does not match.
[WARNING] Execution data for class org/eclipse/hono/util/RequestResponseApiConstants does not match.
[WARNING] Execution data for class org/eclipse/hono/config/ClientConfigProperties does not match.
[WARNING] Execution data for class org/eclipse/hono/util/TriTuple does not match.
[WARNING] Execution data for class org/eclipse/hono/util/TenantObject does not match.
[WARNING] Execution data for class org/eclipse/hono/auth/BCryptHelper does not match.
[WARNING] Execution data for class org/eclipse/hono/tracing/MultiMapExtractAdapter does not match.
[WARNING] Execution data for class org/eclipse/hono/config/PemReader$Entry does not match.
[WARNING] Execution data for class org/eclipse/hono/util/EventBusMessage does not match.
[WARNING] Execution data for class org/eclipse/hono/tracing/JsonObjectExtractAdapter does not match.
[WARNING] Execution data for class org/eclipse/hono/util/PortConfigurationHelper does not match.
[WARNING] Execution data for class org/eclipse/hono/tracing/MessageAnnotationsExtractAdapter$1 does not match.
[WARNING] Execution data for class org/eclipse/hono/tracing/JsonObjectInjectAdapter does not match.
[WARNING] Execution data for class org/eclipse/hono/auth/AuthoritiesImpl does not match.
[WARNING] Execution data for class org/eclipse/hono/auth/EncodedPassword does not match.
[WARNING] Execution data for class org/eclipse/hono/util/Hostnames does not match.
[WARNING] Execution data for class org/eclipse/hono/util/MessageTap does not match.
[WARNING] Execution data for class org/eclipse/hono/auth/Activity does not match.
[WARNING] Execution data for class org/eclipse/hono/util/JwtHelper$1 does not match.
[WARNING] Execution data for class org/eclipse/hono/util/HexDecodingException does not match.
[WARNING] Execution data for class org/eclipse/hono/util/CredentialsConstants does not match.
[WARNING] Execution data for class org/eclipse/hono/util/CredentialsObject does not match.
[WARNING] Execution data for class org/eclipse/hono/util/JwtHelper does not match.
[WARNING] Execution data for class org/eclipse/hono/config/FileFormat does not match.
[INFO] 
[INFO] --- maven-checkstyle-plugin:3.0.0:check (checkstyle-check) @ hono-core ---
[INFO] 
[INFO] --- maven-install-plugin:2.5.2:install (default-install) @ hono-core ---
[INFO] Installing /home/jreimann/Development/workspace-hono/hono/core/target/hono-core-1.0-M2-SNAPSHOT.jar to /home/jreimann/.m2/repository/org/eclipse/hono/hono-core/1.0-M2-SNAPSHOT/hono-core-1.0-M2-SNAPSHOT.jar
[INFO] Installing /home/jreimann/Development/workspace-hono/hono/core/pom.xml to /home/jreimann/.m2/repository/org/eclipse/hono/hono-core/1.0-M2-SNAPSHOT/hono-core-1.0-M2-SNAPSHOT.pom
[INFO] Installing /home/jreimann/Development/workspace-hono/hono/core/target/hono-core-1.0-M2-SNAPSHOT-sources.jar to /home/jreimann/.m2/repository/org/eclipse/hono/hono-core/1.0-M2-SNAPSHOT/hono-core-1.0-M2-SNAPSHOT-sources.jar
[INFO] 
[INFO] --- maven-bundle-plugin:3.5.0:install (default-install) @ hono-core ---
[INFO] Installing org/eclipse/hono/hono-core/1.0-M2-SNAPSHOT/hono-core-1.0-M2-SNAPSHOT.jar
[INFO] Writing OBR metadata
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  6.942 s
[INFO] Finished at: 2019-04-12T11:47:14+02:00
[INFO] ------------------------------------------------------------------------
➜  core git:(feature/fix_javadoc_1) 

~~~

Reply:
Anyway, I assume you prefer it to be in the "management" section, so I updated the commit to move it there.